### ISSUE TITLE
Enable parallel processing of Actions

### ISSUE THREAD

**Issue by Skejven** [Knotx/knotx-fragments-handler#8]
**Is your feature request related to a problem? Please describe.**
Enable parallel processing of `Actions`.
Current implementation enables running only a single `Action` at a time.
To explain the idea better let's consider a case, when we have a user's bookshelf page.
This page, has a component that requires data from 3 external sources:
- `books`,
- `authors`,
- `labels`.

They are totally independent data sources and can be called with any order or simultaneously.
Current actions graph (`tasks`) implementation enables only calling it sequentially:
```hocon
tasks {
  bookshelf {
    action = auth
    onTransition {
      _success {
        action = books
        onTransition {
          _success {
            action = authors
            onTransition {
              _success {
                action = labels
                onTransition {
                  _success {
                    _success {
                      action = te
                    }
                    _error {
                      action = apply-pdp-fallback
                    }
                  }
                  _error {
                    action = apply-labels-fallback
                  }
                }
              }
              _error {
                action = apply-authors-fallback
              }
            }
          }
          _error {
            action = apply-books-fallback
          }
        }
      }
      _error {
        action = not-auth-fallback
      }
    }
  }
}
```
```
actions {
  auth {
    factory = jwt
  }
  books {
    factory = http
    config {
      // endpoint configs here
    }
  }
  authors {
    factory = http
    config {
      // endpoint configs here
    }
  }
  ...
}  
```

**Describe the solution you'd like**
It would be more elegant and (time) efficient to have possibility to call `books`, `authors` and `labels` sources in the same time and configure it as an array of actions:
```hocon
tasks {
  bookshelf {
    action = auth
    onTransition {
      _success {
        action = [
          {
            action = books
            onTransition {
              _error {
                action = apply-books-fallback
              }
            }
          },
          {
            action = authors
            onTransition {
              _error {
                action = apply-authors-fallback
              }
            }
          },
          {
            action = labels
            onTransition {
              _error {
                action = apply-labels-fallback
              }
            }
          }
        ]
        onTransition {
          _success {
            action = te
          }
          _error {
            action = apply-pdp-fallback
          }
        }
      }
      _error {
        action = not-auth-fallback
      }
    }
  }
}
```

Each item of `action` array (simultaneous action) could create its own subgraph.
It seems that there should be some constraint when actions are executed in parallel (array of actions). In that case the only possible transitions would be:
- `_success` when all parallel subgraphs ended with `_success` (no matter applied by the main Action or its fallback),
- `_error` when at least one subgraph of Actions failed (ended with `_error` and was not rescued).

**Describe alternatives you've considered**
Alternative would be using Knot.x 1.X module [Data Bridge](https://github.com/Knotx/knotx-data-bridge) to do any more complicated operations (e.g. parallel processing).
However, approach presented above is more flexible and can be changed easily by just adjusting the configuration.
Additionally, this simple config enables easier applying cache or circuit breaker behaviours for the actions (simple `http`), which will not require implementing it in the Data Bridge module.
Moreover, Data Bridge uses [Adapters](https://github.com/Knotx/knotx-data-bridge#data-sources-adapters) that are registered on the Event Bus. That means, external service data is sent twice between Adapter and Fragment's Handler (Adapter -> Data Bridge Knot -> Handler) and causes more network traffic and serialising/deserialising external source response data.

**Additional context**
If this feature will be implemented, together with simple `http` action factory, Knot.x Data Bridge module will be no longer needed in a form it existed in Knot.x 1.X. 
The `http` Action would behave similar to Data Bridge [Adapters](https://github.com/Knotx/knotx-data-bridge#data-sources-adapters).
